.github/workflows/coverage.yml

```diff
@@ -0,0 +1,76 @@
+name: Coverage
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  run_coverage:
+    name: Coverage
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {
+            name: "Coverage for sc-dictionary fs-storage",
+            os: ubuntu-latest,
+            file_memory: "Dictionary",
+            build_type: "Debug",
+            cc: "gcc",
+            cxx: "g++",
+          }
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        id: install_deps
+        run: scripts/install_deps_ubuntu.sh
+
+      - name: Restore build caches
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          key: ${{ github.job }}-${{ matrix.config.os}}-${{ matrix.config.cxx }}-${{ matrix.config.build_type }}-${{ matrix.config.file_memory }}
```

Comment:
`-g` flags for coverage and debug/release/sanitizer builds differ, so you'll have to add the step name prefix (sanitizer/coverage/build) to the cache key. Otherwise the cache won't be very effective, because using different compile flags may invalidate parts of the cache, making the builds slower.